internal/config/types.go

```diff
@@ -123,17 +125,63 @@ func (c *Config) HasSecret(secret string) bool {
 type Environment = map[string]string
 
 type Python struct {
-	Version        string `toml:"version" json:"version"`
+	Version        string `toml:"version,omitempty" json:"version"`
 	PackageFile    string `toml:"package_file,omitempty" json:"packageFile"`
 	PackageManager string `toml:"package_manager,omitempty" json:"packageManager"`
 }
 
+func (p *Python) FillDefaults(
+	pythonInterpreter *interpreters.PythonInterpreter,
+) {
+	if p != nil && pythonInterpreter != nil && (*pythonInterpreter).IsPythonExecutableValid() {
```

Comment:
Since you are already within the struct method, I don't think you need to check `p != nil`

Reply:
Did not know that!